doc/source/train/user-guides/checkpoints.rst

```diff
@@ -309,6 +316,77 @@ Lower-performing checkpoints are deleted to save storage space. By default, all
     please ensure that the metric is always reported together with the checkpoints.
 
 
+.. _train-validating-checkpoints:
+
+Validating checkpoints asynchronously
+-------------------------------------
+
+You can also asynchronously validate checkpoints that you :func:`~ray.train.report` as follows:
+
+* Define your own ``validate_fn`` that takes a :class:`~ray.train.Checkpoint` to validate
+  and an optional ``validate_config`` dictionary, which contain arguments needed for validation
+  such as the validation dataset, and returns a dictionary of metrics from the validation.
+  We recommend performing the validation with an eval-only :ref:`Trainer <train-overview-trainers>` or with
+  :func:`~ray.data.Dataset.map_batches` - see :ref:`train-validate-fn` for more details.
+* Call :func:`~ray.train.report` with your ``validate_fn`` and a ``validate_config`` dict.
+  Ray Train runs your ``validate_fn`` with the ``validation_config`` and ``checkpoint``
+  in a new Ray task. When that task completes, Ray Train associates the metrics returned by
+  ``validate_fn`` with that ``checkpoint``.
```

Comment:
This is a bit heavy on the text. Consider nested bullet points, individual headers, or putting more information in the API docstrings themselves.